src/Service/MigrationService.php

```diff
@@ -353,6 +360,45 @@ public function findMigrationPath(): string
             : $this->getNamespacePath($this->createNamespace);
     }
 
+    /**
+     * Filters migrations by namespaces and paths.
+     *
+     * @param string[] $classes Migration classes to be filtered.
+     * @param string[] $namespaces Namespaces to filter by.
+     * @param string[] $paths Paths to filter by.
+     *
+     * @return string[] Filtered migration classes.
+     *
+     * @psalm-param list<class-string> $classes
+     *
+     * @psalm-return list<class-string>
+     */
+    public function filterMigrations(array $classes, array $namespaces, array $paths = []): array
+    {
+        $result = [];
+
+        foreach ($paths as $path) {
+            $pathNamespace = $this->getNamespaceFromPath($path);
+
+            if ($pathNamespace !== null) {
+                $namespaces[] = $pathNamespace;
+            }
+        }
+
+        $namespaces = array_map(static fn ($namespace) => trim($namespace, '\\'), $namespaces);
```

Comment:
Need test namespaces when last segment of one namespace is prefix for another.
Example:

```
App\Migrations\User
App\Migrations\UserAccount
```